### PR TITLE
MSCNG: adopt key certificate

### PR DESCRIPTION
**Pull request by vmiklos**
This implements handling of private key / key certificate. Next TODO will be xmlSecMSCngKeyDataX509XmlWrite(), i.e. XML serialization of all this.

### CODE REVIEW

File: src/mscng/app.c [hunk 1]
```diff
@@ -169,14 +186,47 @@ xmlSecMSCngAppKeyCertLoadMemory(xmlSecKeyPtr key, const xmlSecByte* data, xmlSec
...
+    ret = xmlSecBufferReadFile(&buffer, filename);
+    if(ret < 0) {
+        xmlSecInternalError2("xmlSecBufferReadFile", NULL, "filename=%s",
+            xmlSecErrorsSafeString(filename));
+        return(NULL);
+    }
+
+    data = xmlSecBufferGetData(&buffer);
+    if(data == NULL) {
+        xmlSecInvalidDataError("xmlSecBufferGetData", NULL);
```
Comment (lsh123) [inline src/mscng/app.c]: This should be xmlSecInternalError

File: src/mscng/app.c [hunk 2]
```diff
@@ -197,11 +247,168 @@ xmlSecKeyPtr
...
+        DWORD dwDataLen = sizeof(DWORD);
+
+        ret = CertGetCertificateContextProperty(cert, CERT_KEY_SPEC_PROP_ID,
+            &dwData, &dwDataLen);
+        if(ret == TRUE) {
+            /* adopt private key */
+            certDuplicate = CertDuplicateCertificateContext(cert);
+            if(certDuplicate == NULL) {
+                xmlSecMSCngLastError("CertDuplicateCertificateContext", NULL);
+                xmlFree(pwdWideChar);
```
Comment (lsh123) [inline src/mscng/app.c]: There are a lot of cleanups happening here.... I would consider using "goto" instead with initializing all the relevant pointers with NULL at the beggining.

File: src/mscng/app.c (hunk 1 above)
Comment (vmiklos) [reply src/mscng/app.c]: Fixed.

File: src/mscng/app.c (hunk 2 above)
Comment (vmiklos) [reply src/mscng/app.c]: I've added a cleanup label and now I deduplicated that code.

**Pull request merged**